3rd-party/class.jetpack-amp-support.php

```diff
@@ -0,0 +1,300 @@
+<?php
+
+/**
+ * Manages compatibility with the amp-wp plugin
+ *
+ * @see https://github.com/Automattic/amp-wp
+ */
+class Jetpack_AMP_Support {
+
+	static function init() {
+		if ( ! self::is_amp_request() ) {
+			return;
+		}
+
+		// carousel
+		add_filter( 'jp_carousel_maybe_disable', '__return_true' );
+
+		// sharing
+		add_filter( 'sharing_enqueue_scripts', '__return_false' );
+		add_filter( 'jetpack_sharing_counts', '__return_false' );
+		add_filter( 'sharing_js', '__return_false' );
+		add_filter( 'jetpack_sharing_display_markup', array( 'Jetpack_AMP_Support', 'render_sharing_html' ), 10, 2 );
+
+		// disable imploding CSS
+		add_filter( 'jetpack_implode_frontend_css', '__return_false' );
+
+		// enforce freedom mode for videopress
+		add_filter( 'videopress_shortcode_options', array( 'Jetpack_AMP_Support', 'videopress_enable_freedom_mode' ) );
+
+		// include Jetpack og tags when rendering native AMP head
+		add_action( 'amp_post_template_head', array( 'Jetpack_AMP_Support', 'amp_post_jetpack_og_tags' ) );
+
+		// Post rendering changes for legacy AMP
+		add_action( 'pre_amp_render_post', array( 'Jetpack_AMP_Support', 'amp_disable_the_content_filters' ) );
+
+		// Add post template metadata for legacy AMP
+		add_filter( 'amp_post_template_metadata', array( 'Jetpack_AMP_Support', 'amp_post_template_metadata' ), 10, 2 );
+	}
+
+	static function admin_init() {
+		// disable Likes metabox for post editor if AMP canonical disabled
+		add_filter( 'post_flair_disable',  array( 'Jetpack_AMP_Support', 'is_amp_canonical' ), 99 );
+	}
+
+	static function init_filter_jetpack_widgets() {
+		if ( ! self::is_amp_request() ) {
+			return;
+		}
+
+		// widgets
+		add_filter( 'jetpack_widgets_to_include', array( 'Jetpack_AMP_Support', 'filter_available_widgets' ) );
+	}
+
+	static function is_amp_canonical() {
+		return function_exists( 'amp_is_canonical' ) && amp_is_canonical();
+	}
+
+	static function is_amp_request() {
```

Comment:
Could we make this function filterable, so other AMP plugins like [this one](https://wordpress.org/plugins/accelerated-mobile-pages/) can leverage the work we've done in Jetpack?

Reply:
Good call. Done.